gateway-js/src/index.ts

```diff
@@ -529,19 +611,28 @@ export class ApolloGateway implements GraphQLService {
   }
 
   private async updateWithSupergraphSdl(
-    result: SupergraphSdlUpdate,
+    result: SupergraphSdlUpdate | string,
   ): Promise<void> {
-    if (result.id === this.compositionId) {
+    if (typeof result === 'string') {
+    } else if (result.id === this.compositionId) {
       this.logger.debug('No change in composition since last check.');
       return;
     }
 
+    const supergraphSdl =
+      typeof result === 'string' ? result : result.supergraphSdl;
+
+    const id =
+      typeof result === 'string'
+        ? createHash('sha256').update(result).digest('hex')
+        : result.id;
```

Comment:
Just to call out, we'd been avoiding this particular Node.js `crypto` module dependency in lieu of:

[This `apollo-graphql` implementation](https://github.com/apollographql/apollo-server/blob/c49640830bdafc4d499ad338ba2b82545ece4749/packages/apollo-server-core/src/utils/createSHA.ts#L8) and [this similar thing in `apollo-server-core`](https://github.com/apollographql/apollo-tooling/blob/d6f3c14ae7680333c75cac01c67a92fad333b063/packages/apollo-graphql/src/utilities/createHash.ts#L0-L1)

Reply:
Also, is this exactly what we do to generate `compositionId` in Studio? Is that where this addition came from? 🤔

Reply:
Thanks, that jogs my memory! I'm not sure how it's done in Studio, but this is simply a way to allow users to update the supergraph without having to come up with their own unique identifiers on update. Open to feedback on this.

Reply:
1c49051